src/main.cpp

```diff
@@ -2018,8 +2020,8 @@ bool AcceptToMemoryPool(
             return false;
         }
 
-        // Check Orchard bundle authorizations.
-        if (!orchardAuth.Validate()) {
+        // Check Orchard bundle authorizations. `orchardAuth` here is known to be non-null
+        if (!orchardAuth.value().Validate()) {
```

Comment:
Hi @nuttycom and @str4d we're getting intermittent `bad-orchard-bundle-authorization` errors when `zingo` users try to send to the Orchard pool from the Sapling pool.

Can you help us understand how we might be failing this "Orchard bundle authorization" check?  Intermittently.   I see some references to networking and DoS (though it seems to be mempool DoS), are we failing a discrimination against SandBlast bundles here?

Reply:
Please open this as a new issue.